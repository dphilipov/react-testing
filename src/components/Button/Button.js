import styles from './Button.module.css';

function Button({label}) {
  return <div data-testid="button" className={styles.btn}>{label}</div>;
}

export default Button;
