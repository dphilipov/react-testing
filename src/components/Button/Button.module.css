.btn {
    border-radius: 4px;
    background-color: darkslateblue;
    padding: 5px 0;
    width: 200px;
    align-self: center;
    color: #fff;
    font-weight: 700;
    cursor: pointer;
}

.btn:hover {
    background-color: darkviolet;
}