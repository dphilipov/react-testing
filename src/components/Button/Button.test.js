import Button from '../Button/Button';
import { render, cleanup, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(() => {
    cleanup();
})

test('should render component', () => {
    render(<Button label="Test Label"></Button>);

    const ButtonElement = screen.getByTestId('button');
    expect(ButtonElement).toBeInTheDocument();
    expect(ButtonElement).not.toHaveTextContent('');
    expect(ButtonElement).toHaveTextContent('Test Label');
})

test('should match snapshot', () => {
    const tree = renderer.create(<Button label="Test Label"></Button>).toJSON();

    expect(tree).toMatchSnapshot();
})