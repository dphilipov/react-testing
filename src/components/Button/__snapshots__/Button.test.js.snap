// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="btn"
  data-testid="button"
>
  Test Label
</div>
`;
